nodes/IfNode.java

```diff
@@ -54,7 +55,22 @@ public String convertToPython() {
 
     @Override
     public boolean validateTree() {
-        return false;
+
+        if(expr.getType() != Types.BOOLEAN){
+            throw(new Exception()); //TODO Semantic Exception
+            return false;
+        }
+        if(!body.validateTree()){
+            return false;
+        }
+
+        for(ElseifNode e : elifLst) {
+            if (!e.validateTree()) {
+                return false;
+            }
+        }
+
+        return elseCase.validateTree();
```

Comment:
please add `elseCase.setHasIfParent(true);` right above the return statement! This will help me with the elseNode validation and checking that every else has an if